move/open_art_market/sources/open_art_market.move

```diff
@@ -124,7 +129,7 @@ module open_art_market::open_art_market {
             id: object::new(ctx),
             artwork_id: object::uid_to_inner(&artwork.id),
             shares,
-            incoming_price: artwork.incoming_price,
+            incoming_price: (artwork.incoming_price / artwork.total_supply) * shares,
```

Comment:
Need to be careful with division here, floating point numbers are not supported in move. Lets discuss how we can handle this

Reply:
Effectively, `(artwork.incoming_price / artwork.total_supply)` is the share price.
Rather than calculating the share price from `incoming_price` and `total_supply`, I suggest we do it the other way around - calculate `incoming_price` from `artwork.share_price` and `artwork.total_supply`.

In other words, remove the `incoming_price` property and add a `share_price` property instead.

Reply:
Yes this makes a lot of sense!